### ISSUE TITLE
[Enhancement] Validate with -o json does not return all the metadata within the JSON Payload

### ISSUE THREAD

**Issue by fabiodouek** [aws-cloudformation/cloudformation-guard#343]
**Is your feature request related to a problem? Please describe.**

Currently when invoking cfn-guard validate --rules <RULES_DIR> --data <DATA_DIR> -o json, some of the relevant metadata is not in the JSON Payload. This scenario is in the scope of CFN Template evaluation. As follows:

The data file is not specified in the JSON payload, therefore if you are running cfn-guard validate against multiple data files, it's not possible to know which result refers to each file.

A side note that when evaluating a non-CFN Template data file e.g: AWS Config, then the JSON result is completely different and it has an JSON attribute called data_from.

**Describe the solution you'd like**

When using -o json, I would like to have a valid JSON object with the relevant metadata, including the data file path.

**Describe alternatives you've considered**

As a workaround, I'm invoking validate with -o json --show-summary all. It then has a reference to the data file outside of the JSON Payload. I've built a cfn post processor, to join the data file in the JSON, but it's an intensive processing depending on the number of evaluations.

**Comment by akshayrane** [comment]
Hi @fabiodouek 

Could you please also specify the version of `cfn-guard` that you are using, the environment you are running it on (OS/version) and the snippets of input and output to better help us troubleshoot this issue?

**Comment by fabiodouek** [comment]
Hi @akshayrane ,

The cfn-guard version is 2.1.3 running on Amazon Linux 2.

This is the current output:
``` 
kms_0123456789012_23.json Status = PASS
PASS rules
kms_key_accounts_allowed.guard/KMS_KEY_ACCOUNTS_ALLOWED   PASS
---
{
  "name": "",
  "metadata": {},
  "status": "PASS",
  "not_compliant": [],
  "not_applicable": [],
  "compliant": [
    "KMS_KEY_ACCOUNTS_ALLOWED"
  ]
}

```

**Comment by akshayrane** [comment]
Thanks Fabio, we'll take a closer look at this issue, in our test environment and get back.

**Comment by joshfried-aws** [comment]
Hi @fabiodouek 

This issue has been addressed as part of  #358 

I am going to go ahead and close this issue, feel free to reopen if need be!
